tests/js/autoAdjustTests.js

```diff
@@ -0,0 +1,94 @@
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($) {
+    "use strict";
+    
+    fluid.registerNamespace("gpii.tests");
+
+    fluid.defaults("gpii.tests.autoAdjustControls", {
+        gradeNames: ["gpii.adjuster.autoAdjust", "autoInit"],
+        model: {
+            autoAdjust: false
+        }
+    });
+    
+    fluid.defaults("gpii.tests.autoAdjust", {
+        gradeNames: ["fluid.test.testEnvironment", "autoInit"],
+        components: {
+            autoAdjuster: {
+                type: "gpii.tests.autoAdjustControls",
+                container: ".gpii-onOffSwitch-container",
+                options: {
+                    strings: {
+                        label: "Auto-adjust",
+                        description: "Preferences will be automatically adjusted based on background noise and ambient light levels.",
+                    }
+                }
+            },
+            autoAdjustTester: {
+                type: "gpii.tests.autoAdjustTester"
+            }
+        }
+    });
+
+    fluid.defaults("gpii.tests.autoAdjustTester", {
+        gradeNames: ["fluid.test.testCaseHolder", "autoInit"],
+        testOptions: {
+            defaultInputStatus: false,
+            newValue: true
+        },
+        modules: [{
+            name: "Auto-adjust adjuster",
+            tests: [{
+                name: "Description text",
+                func: "gpii.tests.autoAdjust.validateText",
+                args: ["{autoAdjuster}.dom.autoAdjustDescription", "{autoAdjuster}.options.strings.description"]
+            }, {
+                name: "Validate autoAdjust",
+                expect: 4,
+                sequence: [{
+                    name: "Description text",
+                    func: "gpii.tests.autoAdjust.validateText",
+                    args: ["{autoAdjuster}.dom.autoAdjustDescription", "{autoAdjuster}.options.strings.description"]
+                }, {
+                    name: "Label text",
+                    func: "gpii.tests.autoAdjust.validateText",
+                    args: ["{autoAdjuster}.dom.headingLabel", "{autoAdjuster}.options.strings.label"]
+                }, {
+                    name: "Verify checkbox state (unChecked)",
+                    func: "gpii.tests.verifyCheckboxState",
+                    args: ["The auto adjust option is not checked by default", "{that}.options.testOptions.defaultInputStatus", "{autoAdjuster}.dom.valueCheckbox"]
+                }, {
+                    func: "{autoAdjuster}.triggerModelChangeOnActivate",
+                    args: ["{autoAdjuster}", "{autoAdjuster}.events.afterRender"]
+                }, {
+                    name: "Verify checkbox state (checked)",
+                    func: "gpii.tests.verifyCheckboxState",
+                    args: ["The auto adjust option is checked", "{that}.options.testOptions.newValue", "{autoAdjuster}.dom.valueCheckbox"]
+                }]
```

Comment:
these last two could be in a sequence, but the assertion should be bound to a modelChanged event. You can do something like this https://github.com/fluid-project/infusion/blob/master/tests/test-core/testTests/js/TestingTests.js#L558-L565 except you can just use a straight listener and args e.g. 

```javascript
}, {
    func: "{autoAdjuster}.triggerModelChangeOnActivate",
    args: ["{autoAdjuster}", "{autoAdjuster}.events.afterRender"]
}, {
    name: "Verify checkbox state (checked)",
    event: "{autoAdjuster}.events.afterRender"
    func: "gpii.tests.verifyCheckboxState",
    args: ["The auto adjust option is checked", "{that}.options.testOptions.newValue", "{autoAdjuster}.dom.valueCheckbox"]
}
```